lib/Internal/Windows/SocketConnector.php

```diff
@@ -365,6 +365,11 @@ public function onProcessConnectTimeout($watcher, Handle $handle)
         \proc_close($handle->proc);
 
         $handle->joinDeferred->fail($error);
+
+        if ($handle->status === ProcessStatus::STARTING) {
```

Comment:
`$handle->status` should always be `STARTING` at that point, so the `if` is unnecessary, no?

Reply:
You might be right, but I'm finding it difficult to prove this to myself, since this is an async callback and I don't know the codebase well. I think this might have been copied from line 260-261, which has a comment "Required, because a process might be destroyed while starting". I'm not sure if that's still true here.

I have to leave this up to you. Would you like me to remove the check?

Reply:
I went ahead and made the change.